utils/prometheus/prometheus.go

```diff
@@ -53,7 +40,7 @@ type Handler struct {
 	Service        string
 	DeploymentType string
 	Labels         map[string]string
-	HTTPClient     *http.Client
+	API            apiv1.API
```

Comment:
I know it's an acronym and go requires acronym to be all-uppercase, but here we are also exposing API ehich I don't like much... Can we rename the field to something like

```suggestion
	_API            apiv1.API
```
or
```suggestion
	prometheusAPI            apiv1.API
```